File: 2021/May/Day 17/tribonacci.py
# Source :https://leetcode.com/problems/n-th-tribonacci-number/submissions/
class Solution:
    def tribonacci(self, n: int) -> int:
        i=2;

        l = [0,1,1]
        while(i<n+1):


            l.append(l[i]+l[i-1]+l[i-2])
            i = i+1
        return(l[n])

File: 2022/May/Day 27/tribonacci.cpp
// Source : https://leetcode.com/problems/n-th-tribonacci-number/submissions/
class Solution {
public:
    int tribonacci(int n) {
        if(n<=1){return n;}
        if(n==2){return 1;}
        int f[n+1];
        f[0]=0;
        f[1]=f[2]=1;

        for(int i=0;i<n-2;i++)
        {
            f[i+3]= f[i+2]+f[i+1] +f[i];
        }

        return f[n];

    }
};
